pandas/_libs/index.pyx

```diff
@@ -74,10 +76,16 @@ cdef ndarray _get_bool_indexer(ndarray values, object val):
                 indexer[i] = is_matching_na(item, val)
 
     else:
-        if util.is_nan(val):
-            indexer = np.isnan(values)
+        if mask is not None:
+            if val is C_NA:
+                indexer = mask == 1
+            else:
+                indexer = (values == val) & ~mask
```

Comment:
do we need to worry about `util.is_nan(val)` here?

Reply:
good point, yes as long as np.nan can be in Float64 we have to check this too